src/generate.ts

```diff
@@ -47,7 +47,20 @@ const sqlTypeToTypescriptType = (
 		}
 	}
 
-	switch (f.dataType) {
+	// Avoid exposing serial nature of Serial fields
+	let dataType = f.dataType;
+	if (
+		dataType === 'Serial' ||
+		(dataType === 'ForeignKey' &&
+			f.references != null &&
+			m.tables[f.references.resourceName].fields.find(
+				(field) => field.fieldName === f.references?.fieldName,
+			)?.dataType === 'Serial')
```

Comment:
This should be a separate thing for in-lining all references rather than having anything specific to "Serial". Since the problem can occur with any sort of references (including concept type references) where the referenced field may have been removed and is independent from the data type of the referenced field